contracts/LargeStorageManager.sol

```diff
@@ -203,4 +203,17 @@ contract LargeStorageManager {
 
         return true;
     }
+
+    function _getChunkHash(bytes32 key, uint256 chunkId) internal view returns (bytes32) {
+        bytes32 metadata = keyToMetadata[key][chunkId];
+        if (metadata.isInSlot()) {
+            bytes memory res = SlotHelper.getRaw(keyToSlots[key][chunkId], metadata);
+            return keccak256(res);
+        } else {
+            address addr = metadata.bytes32ToAddr();
+            bytes32 hash;
+            assembly { hash := extcodehash(addr) }
```

Comment:
This will not return the data hash as each data contains a small portion of code to self-destruct the contract.

Reply:
The data related to self destruct will be added in the web3q-deployer script, generate hash with added data

Reply:
Yes, but in case we are using a different code, the web3q-deployer will be broken.  E.g., we may use a simpler assembly code to self-destruct.

Reply:
so web3q-deployer needs to be synchronized with the contract

Reply:
I don't think the requirement of sync is a good design.  In general, web3q-deployer should work with any contract implementing W3RC-3 directory standard, no matter what the underlying implementation is.

Reply:
Dynamically return header information